tools/raisinbread_refresh.php

```diff
@@ -0,0 +1,216 @@
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+$info = <<<INFO
+This script is used by LORIS developers to empty their databases and replace
+them with new test data.
+
+
+INFO;
+
+echo $info;
+
+$cwd = getcwd();
+if (substr_compare($cwd, 'tools', mb_strlen($cwd) - mb_strlen('tools')) !== 0) {
+    die('Please run this script from the tools/ directory.' . PHP_EOL);
+}
+
+try {
+    require_once 'generic_includes.php';
+
+    $dbInfo = $config->getSettingFromXML('database');
+
+    if (! $config->getSetting('dev')['sandbox']) {
+        throw new \LorisException(
+            "Config file indicates that this is not a sandbox. Aborting to " .
+            "prevent accidental data loss."
+        );
+    }
+    $dbname = $dbInfo['database'];
+    $host = $dbInfo['host'];
+    $username = $dbInfo['username'];
+    $password = $dbInfo['password'];
+
+    $urlConfigSetting = $config->getSetting('url');
+    $baseConfigSetting = $config->getSetting('base');
+    $hostConfigSetting = $config->getSetting('host');
+
+} catch (\DatabaseException $e) {
+    echo 'Could not connect to the database in the Config file.' .
+        'It\'s possible that the LORIS tables have already been dropped.' .
+        PHP_EOL;
+    echo 'Continue attempting to install Raisinbread database? (yN)' . PHP_EOL;
+    $input = trim(fgets(STDIN));
+
+    if (mb_strtolower($input) !== 'y') {
+        die;
+    }
+    echo PHP_EOL . 'Please enter the name of your database:' . PHP_EOL;
+    $dbname = trim(fgets(STDIN));
+} catch (\LorisException $e) {
+    die($e->getMessage());
+} catch (Exception $e) {
+    die("Could not load project/config.xml");
+}
+
+
+echo <<<CONFIRMATION
+Please re-type the database name `$dbname` to confirm you wish to drop tables
+and import test data: 
+CONFIRMATION;
+
+$input = trim(fgets(STDIN));
+if ($input !== $dbname) {
+    die('Input did not match database name. Exiting.');
+}
+
+// Create a connection to mysql via bash. All credentials are supplied via 
+// command line arguments except for the password which must be entered manually.
+$mysqlCommand = "mysql -A \"$dbname\"";
+
+// Drop tables
+echo PHP_EOL .'Dropping LORIS tables....' . PHP_EOL;
+dropTables();
+
+
+// Source LORIS core tabes
+echo <<<INFO
+[*] Creating LORIS core tables.... 
+
+INFO;
+$coreTables = glob("../SQL/0000-*.sql"); 
+array_walk($coreTables, 'runPatch');
+
+// Create instrument tables
+echo <<<INFO
+[*] Creating instrument tables.... 
+
+INFO;
+$rbInstrumentTables = glob("../raisinbread/instruments/instrument_sql/*.sql"); 
+array_walk($rbInstrumentTables, 'runPatch');
+
+// Import Raisinbread data
+echo <<<INFO
+[*] Importing Raisinbread data....
+
+INFO;
+$rbData= glob("../raisinbread/RB_files/*.sql");
+array_walk($rbData, 'runPatch');
+
+// Restore config settings if they were successfully found before.
+if (isset($urlConfigSetting)) {
```

Comment:
scope of $urlConfigSetting different than the try-catch block.

Reply:
I don't think try-catch has a different scope actually. It's not the same as declaring a new variable in a for-loop or a new function anyway.

The code runs as written and will restore the config setting.